Jenkinsfile

```diff
@@ -1,14 +1,69 @@
+def nexusSnapshotRepoDetails = getNexusSnapshotRepoDetails()
+def nexusReleaseRepoDetails = getNexusReleaseRepoDetails()
+def mavenSettingsId = getMavenSettingsId()
+def mavenAdditionalSettings = getMavenAdditionalSettings()
+def enceladusSlaveLabel = getEnceladusSlaveLabel()
+def deploymentStageRun = 'false'
+
 pipeline {
-     agent {
-         docker {
-             image 'maven:3-alpine'
-         }
-     }
-     stages {
-         stage('Build') {
-             steps {
-                 sh 'mvn -B -DskipTests clean install'
-             }
-         }
-     }
- }
+    agent {
+        label "${enceladusSlaveLabel}"
+    }
+    tools {
+        jdk 'openjdk-1.8.0'
+        maven 'Maven-3.6.0'
+        git 'git-latest'
+    }
+    options {
+        buildDiscarder(logRotator(numToKeepStr: '20'))
+        timestamps()
+    }
+    stages {
+        stage ('Prepare') {
+            steps {
+                enceladusCreateConfigurationFiles()
+            }
+        }
+        stage ('Build') {
+            steps {
+                configFileProvider([configFile(fileId: "${mavenSettingsId}", variable: 'MAVEN_SETTINGS_XML')]) {
+                    sh "mvn -s $MAVEN_SETTINGS_XML clean package ${mavenAdditionalSettings}"
+                }
+            }
+        }
+        stage ('Deploy-to-Snapshot') {
```

Comment:
Having these deploy as two separate stages looks bad in the pipeline view. I would put it in one stage.

Reply:
That was my original plan, but using more than one 'when' within a stage doesn't work; Jenkins will complain.
If you know of any other options I'm happy to take a look.

Reply:
Updated to have one stage for build, resolving.